examples/stereo_reconstruction_core_and_direction.py

```diff
@@ -0,0 +1,494 @@
+"""Example of showers reconstruction."""
```

Comment:
It also duplicates much of the `ImageProcessor`, why not just use that? This is not good practive.

I would for now just remove this example.

Reply:
The example is the one that makes the plots you see in description and which I put also in the documentation

It replicates `ImageProcessor` because it was made before that class existed

Reply:
Can we minimize it so that it *only* produces that plot without any interactivity? And probably use the `..plot` feature we spoke about so we don't store the png but create the plot in the docs build.

Reply:
I can't simplify it too much because I still need to convert from CameraFrame to TelescopeFrame so I have to expose the event/tel_id loops

Reply:
the interactivity is just there to allow a user to play with it with different test files and exit the event loop easily

Reply:
I understand this point. But then we should have two things, the code that produces only the example plot in the documentation and that fast and quick without interactivity and maybe this example.

However, I specifically dislike this example because:

a) It mixes unrelated things (signal handling, stereo reconstruction, configuration, ...)
b) It teaches people wrong things, e.g. the use of argparse to set things that should just be configurable via the config system. So please, if you make an example that should take cli arguments, use `Tool`.
c) the example has a double function as the code that creates a documentation plot. This is not good. It should be decoupled and the plot in the docs should be recreated during the docs build using a `.. plot::` block.

Reply:
a) I can take out the signal handling but then it's a bit ugly to use; how can I split stereo reco and configuration? configuration is needed for practically everything
b) the Tool class didn't exists when I did this...plus, does it make sense to make an entire tool for a couple of Cl args? it's just an example in the end
c) the problem is how to fix an event from a test file that could disappear tomorrow - the plot in the docs is not meant to be reproducible, just nice to see (any event with e.g. E_true > 5 TeV and 4/5 telescopes makes a nice plot)
d) I can't use ImageProcessor because I need to clean and parameterize in the TelescopeFrame (but also only parameterized works)

Reply:
> c) the problem is how to fix an event from a test file that could disappear tomorrow 

No test file can "just disappear tomorrow". If we remove a test file we rely on from the dataserver, all code using that has to be adapted. that is not an argument.

> b) plus, does it make sense to make an entire tool for a couple of Cl args? it's just an example in the end

A simple tool does not take more lines than your argparse parser here and even avoids duplicated definition of some of the options since you can properly use our config system instead of hand coding options in argparse that are then given to the config system. It just shows bad practice in an example, which is something we must avoid.

d) Then we should add an option to the image processor to do it in TelescopeFrame and add the example when this is done. 

I just don't think this is a good example, especially in the light of the stage2 tool basically doing the same but right.

Reply:
do I remove it then?

Reply:
removed